url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/agent/NestAgent.java

```diff
@@ -0,0 +1,363 @@
+package agent;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Semaphore;
+
+import DeviceGraphics.DeviceGraphics;
+import DeviceGraphics.PartGraphics;
+import GraphicsInterfaces.NestGraphics;
+import agent.data.Part;
+import agent.interfaces.Nest;
+import factory.PartType;
+
+/**
+ * Nests hold parts
+ * 
+ * @author Arjun Bhargava, Michael Gendotti
+ */
+public class NestAgent extends Agent implements Nest {
+
+	public List<PartType> requestList = Collections.synchronizedList(new ArrayList<PartType>());
+	PartType currentPartType = null;
+	public List<MyPart> currentParts = Collections.synchronizedList(new ArrayList<MyPart>());
+	public int count = 0;
+	public int countRequest = 0;
+	int full = 8;
+	public boolean takingParts = false;
+	private boolean partReady = false;
+	private final boolean partTypeNull = false;
+
+	public NestGraphics nestGraphics;
+	private NestState state;
+
+	public Semaphore animation = new Semaphore(0, true);
+
+	String name;
+
+	LaneAgent lane;
+	LaneState laneState = LaneState.READY;
+	CameraAgent camera;
+
+	public class MyPart {
+		Part part;
+		NestStatus status;
+
+		public MyPart(Part p) {
+			part = p;
+			status = NestStatus.IN_NEST;
+		}
+	}
+
+	public enum NestStatus {
+		IN_NEST, IN_NEST_POSITION, REMOVING
+	};
+
+	public enum NestState {
+		PURGING, PRIORITY_PURGE, WAITING_FOR_LANE_PURGE, DONE_PURGING, NULL
+	};
+
+	public enum LaneState {
+		PURGING, READY
+	}
+
+	public NestAgent(String name) {
+		super();
+		state = NestState.DONE_PURGING;
+		this.name = name;
+	}
+
+	// MESSAGES
+	@Override
+	public void msgHereIsPartType(PartType type) {
+		print("Received msgHereIsPartType");
+		if (currentPartType == null && type == null) {
+			currentPartType = type;
+		}
+		else if(currentPartType == null && type != null) {
+			//if(!type.equals(currentPartType)) {
+				currentPartType = type;
+				state = NestState.PURGING;
+				laneState = LaneState.PURGING;
+				lane.msgPurgeParts();
+			//} 
+		}
+		else if(currentPartType != null && type != null) {
+			if(!type.equals(currentPartType)) {
+				currentPartType = type;
+				state = NestState.PURGING;
+				laneState = LaneState.PURGING;
+				lane.msgPurgeParts();
+			} else {
+				state = NestState.DONE_PURGING;
+				takingParts=false;
+			}
+		} else if (currentPartType != null) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of != on line 96 to == check before setting the field value on line 97 or verify if the check is needed at all.